lib/ide/src/view/text_editor.rs

```diff
@@ -0,0 +1,118 @@
+//! This module contains TextEditor, an UiComponent to edit Enso Modules or Text Files.
+
+use crate::prelude::*;
+
+use super::ui_component::UiComponent;
+use super::ui_component::Padding;
+
+use basegl::display::object::DisplayObjectOps;
+use basegl::display::shape::text::glyph::font::FontRegistry;
+use basegl::display::shape::text::text_field::TextField;
+use basegl::display::shape::text::text_field::TextFieldProperties;
+use basegl::display::world::*;
+
+use nalgebra::Vector2;
+use nalgebra::zero;
+
+
+
+// =============
+// === Color ===
+// =============
+
+//TODO[dg]:Move this to a better place.
+mod color {
+    use nalgebra::Vector4;
+
+    pub type Color = Vector4<f32>;
+
+    pub fn black() -> Color { Vector4::new(0.0, 0.0, 0.0, 1.0) }
+}
+
+
+
+
+// ==================
+// === TextEditor ===
+// ==================
+
+/// TextEditor allows us to edit text files or Enso Modules. Extensible code highlighting is
+/// planned to be implemented for it.
+#[derive(Clone,Debug)]
+pub struct TextEditor {
+    text_field : TextField,
+    padding    : Padding,
+    position   : Vector2<f32>,
+    dimensions : Vector2<f32>
+}
+
+impl TextEditor {
+    /// Creates a new TextEditor.
+    pub fn new(world:&World) -> Self {
+        let scene        = world.scene();
+        let camera       = scene.camera();
+        let screen       = camera.screen();
+        let mut fonts    = FontRegistry::new();
+        let font         = fonts.get_or_load_embedded_font("DejaVuSansMono").unwrap();
+        let padding      = default();
+        let position     = zero();
```

Comment:
This initialization of `TextField` is huge. @farmaazon you haven't make it default to some valeus yet? Will it be done in next PR?

Reply:
No, I told you, that this requires a bigger task. And I can create it, but I though that want to work on integration as soon as possible?